entity-service/src/main/kotlin/com/egm/stellio/entity/repository/Neo4jRepository.kt

```diff
@@ -84,17 +84,55 @@ class Neo4jRepository(
         return session.query(query, parameters).queryStatistics().propertiesSet
     }
 
-    fun updateEntityAttribute(entityId: URI, propertyName: String, propertyValue: Any): Int {
+    fun partialUpdatePropertyOfAttribute(
+        attributeId: URI,
+        propertyName: String,
+        propertyPayload: Map<String, Any?>,
+        datasetId: URI? = null
+    ): Int {
+        val matchQuery = if (datasetId == null)
+            """
+            MATCH ({ id: ${'$'}attributeId })-[:HAS_VALUE]->(property:Property { name: ${'$'}propertyName })
+            WHERE NOT EXISTS (property.datasetId)
+            """.trimIndent()
+        else
+            """
+            MATCH ({ id: ${'$'}attributeId })-[:HAS_VALUE]
+            ->(property:Property { name: ${'$'}propertyName, datasetId: ${'$'}datasetId })
+            """.trimIndent()
+
+        val partialUpdateQuery =
+            """
+            SET property += ${'$'}props,
+                property.modifiedAt = datetime("${Instant.now().atZone(ZoneOffset.UTC)}")
```

Comment:
I remember seeing recent comments about the useless of calling `(...).atZone(ZoneOffset.UTC)`, no?

Reply:
I did comment about this, but removed the comments. I think when we need a ZonedDateTime it's ok to have Instant.now().atZone(ZoneOffset.UTC).But here we need a string representation, so Instant.now() with the implicit toString() call is better